lib/Doctrine/ORM/Cache/DefaultCacheFactory.php

```diff
@@ -178,6 +181,19 @@ public function buildQueryCache(EntityManagerInterface $em, $regionName = null)
      */
     public function buildCollectionHydrator(EntityManagerInterface $em, array $mapping)
     {
+        if ($mapping['cache']) {
+            $targetPersister = $em->getUnitOfWork()->getEntityPersister($mapping['targetEntity']);
+
+            if ($targetPersister instanceof CachedPersister) {
```

Comment:
I really hate this nested `if`... but I do not know why ` $em->getUnitOfWork()->getEntityPersister($mapping['targetEntity']);` seems to be instance of `BasicEntityPersister`.... and not a cache persister as expected...